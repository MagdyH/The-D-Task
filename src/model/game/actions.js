

import ActionType from './constants';

export const Actions = {
    getAllGames: (games)=>{
        return {type:ActionType.GET_ALL_GAMES,games}
    },
    getGameBetweenTwoClubs: (games)=>{
        return {type:ActionType.GET_GAME_BETWEEN_TWO_CLUBS,games}
    }
}
export default Actions;
