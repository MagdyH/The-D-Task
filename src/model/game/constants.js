export const ActionType = {
    GET_ALL_GAMES: 'game/GET_ALL_GAMES',
    GET_GAME_BETWEEN_TWO_CLUBS:'game/GET_GAME_BETWEEN_TWO_CLUBS'
}

export default ActionType;