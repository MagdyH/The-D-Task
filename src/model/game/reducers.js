import ActionType from './constants';

export default function gameReducer(state = { games: [],storedGames:[] }, action) {
    switch (action.type) {
        case ActionType.GET_ALL_GAMES:
            return { ...state, games: action.games,storedGames:[] }
        case ActionType.GET_GAME_BETWEEN_TWO_CLUBS:
                
            let games =action.games ? action.games.firstTeam_VS_secondTeam:[];
            return { ...state, games: games }
        default:
            return state;
    }
}