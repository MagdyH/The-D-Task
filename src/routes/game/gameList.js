import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ClubCard from '../../components/clubCard';
import API from '../../api/footballAPI';
import Actions from '../../model/club/actions';
import GameCard from '../../components/gameCard';
import Utilities from '../../utilities/utility';

/*
container component to show games list and teams.
*/
export function GameList(props) {
    let list = [];
    let latestGames = [];
    let sortedGames = [];
    var date = new Date();
    if(props.Actions){
        props.Actions.applySearchMode('Games');
    }

    if (props.clubs && props.clubs.length > 0) {
        list = props.clubs.map((club) => <ClubCard key={club.team_key} club={club} />)
    }

    if(props.games && props.games.length > 0)
    {
        latestGames = props.games.filter((g) => Utilities.convertDate(g.match_date) <= Utilities.convertDate(date));
        sortedGames = latestGames.sort(function (a, b) {
            return new Date(a.match_date) - new Date(b.match_date);
        }).reverse();
    }

    var gamesGroupByDate = sortedGames.reduce((games, game) => {
        if (!games[game.match_date]) games[game.match_date] = [];
        games[game.match_date].push(game);
        return games;
    }, {});

    const gamesGroup = Object.keys(gamesGroupByDate).map((date) => {
        var date1 = new Date(date);
        date1.setHours(0,0,0,0);
        var date2 = new Date();
        date2.setHours(0,0,0,0);

        var stringDate;
        if(Date.parse(date1) === Date.parse(date2))
        {
            stringDate = "Today";
        }
        else{
            stringDate = date1.toDateString();
        }
        return {
            stringDate,
            games: gamesGroupByDate[date]
        };
    });
    return (
        <div className="container club-data">
            <div className="row my-5">
                <div className="col-md-9">
                    {gamesGroup.length > 0 &&
                        gamesGroup.map((group,index) => <div key={index}>
                            <h1>{group.stringDate}</h1>
                            {group.games.map((game) => <div key={game.match_id}> <GameCard key={game.match_id} game={game} /></div>)}
                        </div>)

                    }
                </div>
                <div className="my-4 col-md-3 Scrollable ">
                    {list}
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state, ownProps) {
    return {
        clubs: state.club ? state.club.clubs:[],
        games: state.game ? state.game.games:[]
    }
}


function mapDispatcherToProps(dispatch) {
    return {
        API: bindActionCreators(API, dispatch),
        Actions: bindActionCreators(Actions, dispatch)
    }

}

export default connect(mapStateToProps, mapDispatcherToProps)(GameList);

