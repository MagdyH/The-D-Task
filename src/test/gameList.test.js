import React from 'react';
import renderer from 'react-test-renderer';
import {GameList as Games} from '../routes/game/gameList';


describe('GameList component', () => {
    it('should render snapshot', () => {
      const component = renderer.create(<Games />);
  
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

});