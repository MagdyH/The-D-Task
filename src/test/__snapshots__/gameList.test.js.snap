// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameList component should render snapshot 1`] = `
<div
  className="container club-data"
>
  <div
    className="row my-5"
  >
    <div
      className="col-md-9"
    />
    <div
      className="my-4 col-md-3 Scrollable "
    />
  </div>
</div>
`;
